docs/designs/011-kubernetes-object-monitor.md

```diff
@@ -0,0 +1,267 @@
+
+## Solution 
+
+Develop a policy-driven Kubernetes health monitor that is configurable via CEL policies and publishes health events based on rule matches. This allows users to integrate existing health checks without writing new health monitors.
+
+
+An example of such configuration would be:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubernetes-health-monitor
+  namespace: nvsentinel
+data:
+  policies.yaml: |
+    policies:
+      # Node condition-based policy
+      # Node not ready for more than 2 hours
+      - name: GPUNodeNotReady
+        enabled: true
+        resourceType: Node
+        predicate: |
+          has(node.metadata.labels['nvidia.com/gpu.present']) &&
+          node.metadata.labels['nvidia.com/gpu.present'] == 'true' &&
+          node.status.conditions.exists(c, 
+            c.type == 'Ready' && 
+            (c.status == 'Unknown' || c.status == 'False')
+          )
+        timeThreshold: 2h
+        healthEvent:
+          componentClass: "Node"
+          isFatal: true
+          message: "GPU node has been NotReady for more than 2 hours"
+          recommendedAction: "REBOOT_NODE"
+      
+      # Event-based policy (GPU container creation failures from Pods)
+      # Note: These are Pod events, but indicate Node-level GPU issues
+      # Events don't support timeThreshold - immediate evaluation
+      - name: NVMLError
+        enabled: true
+        resourceType: Event
+        predicate: |
+          event.type == 'Warning' &&
+          event.reason == 'Failed' &&
+          event.involvedObject.kind == 'Pod' &&
+          event.source.component == 'kubelet' &&
+          event.message.contains('nvidia-container-cli') &&
+          event.message.contains('nvml error') &&
+          event.count > 2
+        healthEvent:
+          componentClass: "GPU"
+          isFatal: true
+          message: "GPU container runtime failures detected - likely node GPU driver issue"
+          recommendedAction: "REBOOT_NODE"
+```
+
+### Design
+
+#### Architecture Overview
+
+The Kubernetes Object Monitor is implemented as a **Deployment** that uses **Kubernetes informers** with a **work-queue** for efficient cluster-wide monitoring. It evaluates CEL-based policies against Kubernetes objects (Nodes, Events) and publishes HealthEvents when policies match.
+
+**Key Components:**
+- **Informer-based watch**: Efficiently monitors Nodes and Events using Kubernetes client-go informers
+- **Work-queue**: Provides retry logic, rate-limiting, and deduplication
+- **CEL Policy Engine**: Evaluates user-defined predicates against Kubernetes objects
+
+
+**Watched Resources:**
```

Comment:
This is a great start, can we keep the door open to generic resources as well? Can we add GVK/GVR and use that construct the informers rather than just Nodes and Events? 

For the initial iterations only Nodes and Events will be supported and tested, but in future we want to add a vendor specific CR, then let's keep the option open for that now rather than having to do a refactor for that later.